runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -167,6 +167,19 @@ outOfProcessCompilationEnd(
 
    Trc_JITServerCompileEnd(compInfoPT->getCompilationThread(), compInfoPT->getCompThreadId(),
          compInfoPT->getCompilation()->signature(), compInfoPT->getCompilation()->getHotnessName());
+
+   // Re-acquire the compilation monitor now that the last message has been sent
+   //
+   J9VMThread *vmThread = compInfoPT->getCompilationThread();
+   TR::CompilationInfo *compInfo = compInfoPT->getCompilationInfo();
+   compInfo->debugPrint(vmThread, "\tacquiring compilation monitor\n");
+   compInfo->acquireCompMonitor(vmThread);
+   compInfo->debugPrint(vmThread, "+CM\n");
+   // Acquire the queue slot monitor now
+   //
+   compInfo->debugPrint(vmThread, "\tre-acquiring queue-slot monitor\n");
+   entry->acquireSlotMonitor(vmThread);
+   compInfo->debugPrint(vmThread, "+AM-", entry);
```

Comment:
I am favoring moving this code at the end of postCompilationTasks() to minimize the time we hold the compilation monitor but also to keep the lock acquisition at the same level (for non-jitserver we acquire the compilation monitor in postCompilationTasks, so we should do it i this method for jitserver as well)